apps/website/src/content/docs/table.mdx

```diff
@@ -8,14 +8,360 @@ thumbnail: Table
+
+For tables with large datasets, the `enableVirtualization` prop can be set to `true` to try to enhance performance by only rendering the rows that are currently in the viewport + some buffer rows.
+
+Having a height on the table is required for virtualization to work.
+
+<LiveExample src='Table.virtualization.jsx'>
+  <AllExamples.TableVirtualizationExample client:load />
+</LiveExample>
+
+### Controlled state / `stateReducer`
+
+The uncontrolled Table state is sufficient for most use-cases. However, for advanced control over the table state, pass a `stateReducer` function. This function receives the `action`, `prevState`, and `newState` and thus allows to run custom logic based on these arguments and return a custom new state.
+
+The `stateReducer` function also provides the Table's `instance`. This allows manually triggering actions on the Table.
+
+```jsx
+const tableInstance = React.useRef<TableInstance<DemoData>>(undefined);
+
+// Example of manually triggering an action
+const toggleRowSelected = (id: string, checked: boolean) => {
+  tableInstance.current?.toggleRowSelected(id, checked);
+}
+
+<Table
+  /* … */
+  stateReducer={
+    useCallback((newState, action, prevState, instance) => {
+      tableInstance.current = instance;
+
+      // Some custom logic depending on action
+      // if (action.type === "…") {
+      //   …
+      // }
+
+      return newState;
+    }, []) satisfies NonNullable<TableOptions<DemoData>['stateReducer']>
+  }
+/>
+```
+
+### Selection
+
+The `Table` component supports row selection, allowing users to select one or more rows for actions by setting the `isSelectable` prop to `true`.
+
+```jsx {6}
+<Table
+  /* … */
+  data={data}
+  columns={columns}
+  emptyTableContent='No data.'
+  isSelectable
+/>
+```
+
+There are two available selection modes passed using the `selectionMode` prop: `"single"` and `"multi"` (default).
+
+#### Single-row selection
+
+In single row selection mode (`selectionMode="single"`), users can select each row by clicking on the desired row.
+
+**Note**: For accessibility reasons, in single-selection Tables, it is highly recommended to include a button in each selectable row. [More info](https://github.com/iTwin/iTwinUI/issues/2130#issuecomment-2209287280).
+
+```jsx
+const columns = React.useMemo(
+  () =>
+    [
+      /* … */
+      {
+        /* … */
+        Cell: ({ value }) => {
+          return (
+            <Anchor
+              as="button"
+              onClick={() => console.log(`Selected ${value}`)}
+            >
+              {value}
+            </Anchor>
+          );
+        },
+      },
+    ] satisfies Column[],
+  []
+);
+```
+
+<LiveExample src='Table.selectSingle.jsx'>
+  <AllExamples.TableSelectSingleExample client:load />
+</LiveExample>
+
+#### Multi-row selection
```

Comment:
Multi-row selection should be documented before single-row selection.

Also we need to mention `selectOnRowClick` somewhere.

Reply:
Typo. I meant `selectRowOnClick`, not `selectOnRowClick`.

https://github.com/iTwin/iTwinUI/blob/1b78680da0b105afaa616d9f289eb99113978946/packages/itwinui-react/src/core/Table/Table.tsx#L433

Reply:
I think it could be a third subheading after "multi-row selection" and "single-row selection"? Multi should definitely be the first one.